### PR TITLE
 ues strategy pattern modify refresh ShardingMetaData.

### PR DESCRIPTION
**Pull request by yu199195**

### CODE REVIEW

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/AbstractTableStatementMetaData.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * The type Abstract table statement meta data.
+ */
+public abstract class AbstractTableStatementMetaData {
+    
+    /**
+     * Load table meta table meta data.
+     *
+     * @param tableName              the table name
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/AbstractTableStatementMetaData.java]: Please reference other javadoc and make them consist

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/DropTableStatementMetaDataRefresh.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ */
+public final class DropTableStatementMetaDataRefresh implements SQLStatementMetaDataRefresh<DropTableStatement> {
+    
+    @Override
+    public void refreshMetaData(final ShardingRuntimeContext shardingRuntimeContext, final SQLStatementContext<DropTableStatement> sqlStatementContext) {
+        for (SimpleTableSegment each : sqlStatementContext.getSqlStatement().getTables()) {
+            shardingRuntimeContext.getMetaData().getSchema().remove(each.getTableName().getIdentifier().getValue());
+        }
+    }
+
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/DropTableStatementMetaDataRefresh.java]: Please remove useless line

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/SQLStatementMetaDataRefreshFactory.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    static {
+        REFRESH_MAP.put(CreateTableStatementContext.class, new CreateTableStatementMetaDataRefresh());
+        REFRESH_MAP.put(AlterTableStatementContext.class, new AlterTableStatementMetaDataRefresh());
+        REFRESH_MAP.put(DropTableStatementContext.class, new DropTableStatementMetaDataRefresh());
+        REFRESH_MAP.put(CreateIndexStatementContext.class, new CreateIndexStatementMetaDataRefresh());
+        REFRESH_MAP.put(DropIndexStatementContext.class, new DropIndexStatementMetaDataRefresh());
+    }
+    
+    /**
+     * New instance sql statement meta data refresh.
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/SQLStatementMetaDataRefreshFactory.java]: `sql` should be `SQL` in javadoc

Comment (yu199195) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/SQLStatementMetaDataRefreshFactory.java]: fix.

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/DropTableStatementMetaDataRefresh.java (hunk 2 above)
Comment (yu199195) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/DropTableStatementMetaDataRefresh.java]: ok，fix it

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/AbstractTableStatementMetaData.java (hunk 1 above)
Comment (yu199195) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/AbstractTableStatementMetaData.java]: fix it

Comment (terrymanu) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/AbstractTableStatementMetaData.java]: It looks like not fixed yet.

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/SQLStatementMetaDataRefresh.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import org.apache.shardingsphere.sql.parser.sql.statement.SQLStatement;
+
+import java.sql.SQLException;
+
+/**
+ * The interface SQL statement meta data refresh.
+ *
+ * @param <T> the type parameter
+ */
+public interface SQLStatementMetaDataRefresh<T extends SQLStatement> {
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/refreh/SQLStatementMetaDataRefresh.java]: Class name should be noun

**Pull request merged**